io/disk/vfc-tests/vfc_unmap_map.py

```diff
@@ -0,0 +1,348 @@
+               vfcclient -fmt :"' % (self.server, vios_name)
+        output = self.run_command(cmd)
+        self.log.info("output value : %s" % output)
+        for line in output:
+            if vfchost in line:
+                return line.split(":")[-1]
+        return ''
+
+    def get_vfchost_status(self, vfchost, vios_name):
+        '''
+        returns the linux_name of corresponding drc_name
+        '''
+        cmd = 'viosvrcmd -m %s -p %s -c "lsmap -all -npiv -field  name\
+               Status -fmt :"' % (self.server, vios_name)
+        output = self.run_command(cmd)
+        self.log.info("output value : %s" % output)
+        for line in output:
+            if vfchost in line:
+                return line.split(":")[-1]
+        return ''
+
+    def get_paths(self, vfc_client):
+        '''
+        returns the mpaths corresponding to linux name
+        '''
+        paths = []
+        cmd = "ls -l /sys/block/"
+        output = process.system_output(cmd).decode('utf-8').splitlines()
+        for line in output:
+            if "/%s/" % vfc_client in line:
+                paths.append(line.split("/")[-1])
+        return paths
+
+    def vfchost_map_unmap(self, operation, vfc_dic):
+        '''
+        Adds and removes a Network virtualized device based
+        on the operation
+        '''
+        self.log.info("%sing %s" % (operation, vfc_dic["vfchost"]))
+        if operation == 'map':
+            cmd = 'viosvrcmd -m %s -p %s -c "vfcmap -vadapter %s -fcp %s"' \
+                   % (self.server, vfc_dic["vios"],
+                      vfc_dic["vfchost"], vfc_dic["fcs"])
+        else:
+            cmd = 'viosvrcmd -m %s -p %s -c "vfcmap -vadapter %s -fcp"' \
+                   % (self.server, vfc_dic["vios"], vfc_dic["vfchost"])
+        try:
+            self.run_command(cmd)
+        except CommandFailed as cmdfail:
+            self.log.debug(str(cmdfail))
+            self.fail("vfchost %s operation failed" % operation)
+        self.vfchost_status_verify(operation, vfc_dic["vfchost"],
+                                   vfc_dic["vios"])
+        self.vfc_client_status_verify(operation, vfc_dic["vfc_client"])
+        self.mpath_verification(operation, vfc_dic["paths"])
+
+    def vfc_client_status_verify(self, operation, vfc_client):
+        '''
+        Returns the WWPNs of give client slot number
+        '''
+        self.log.info("verifying %s status after %s its \
+                       vfchost" % (vfc_client, operation))
+
+        def is_host_online():
+            file_name = '/sys/class/fc_host/%s/port_state' % vfc_client
+            status = genio.read_file(file_name).strip("\n")
+            if operation == "map":
+                if status == 'Online':
+                    return True
+                return False
+            elif operation == 'unmap':
+                if status == 'Linkdown':
+                    return True
+                return False
```

Comment:
Why is this needed inside the function vfc_client_status_verify() ?

Reply:
this is to avoid the waiting time that we use to have with time.sleep

Reply:
@Naresh-ibm My question was: "Why is this needed inside the function vfc_client_status_verify() ?"